diesel/src/expression/array_comparison.rs

```diff
@@ -74,15 +72,15 @@ impl<T, U> NotIn<T, U> {
 impl<T, U> Expression for In<T, U>
 where
     T: Expression,
-    U: Expression<SqlType = T::SqlType>,
+    U: InExpression<SqlType = T::SqlType>,
 {
     type SqlType = Bool;
```

Comment:
TODO: this seems wrong: PG returns null for IN expressions if LHS is null

Reply:
Yes indeed that's already wrong on master as showcased by 6912d7b23d4433ae5dc03afce3367a1de966a69c

Reply:
#4354